Lab_2_3_4_5/src/Grammar/Grammar.java

```diff
@@ -34,7 +34,7 @@ public Grammar(String filePath) throws Exception {
             }
             if(currentLineNumber >= 4) {
                 var lineProductions = Arrays.stream(line.split(";")).collect(Collectors.toList());
-                productionList = Stream.concat(productionList.stream(), lineProductions.stream()).collect(Collectors.toList());
+                productionListAsString = Stream.concat(productionListAsString.stream(), lineProductions.stream()).collect(Collectors.toList());
             }
             currentLineNumber = currentLineNumber + 1;
         }
```

Comment:
I see that TransformProductionStringListToMap throws an exception and it is not caught